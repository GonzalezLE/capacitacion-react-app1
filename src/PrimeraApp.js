import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo})=>{
  return(
    <>
      <h1>{saludo}</h1>
      <p>{subtitulo} </p>
    </>
  )
}
PrimeraApp.propTypes={
  saludo: PropTypes.number.isRequired
};
PrimeraApp.dedaultProps={
  subtitulo:'soy un sub titulo'
}
export default PrimeraApp;