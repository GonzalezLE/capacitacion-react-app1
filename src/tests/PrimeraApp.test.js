import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import PrimeraApp from './../PrimeraApp';


describe('Pruebas con <PrimeraApp />', () => {
  test('debe de mostrar <PrimeraApp /> correctamente', () => {
    const saludo="Hola soy ,luis";
    const wraper =shallow(<PrimeraApp saludo={saludo}/>)
    expect(wraper).toMatchSnapshot();
  });

  test('debe de mostrar el subtitulo mostrado por prop',()=>{
    const saludo = "Hola soy ,luis";
    const sub='soy el sub titulo..!';
    const wraper = shallow(
                      <PrimeraApp 
                        saludo={saludo} 
                        subtitulo={sub}  
                      />);

    //expect(wraper).toMatchSnapshot();
    //document.querySelector()
    const textoParrafo=wraper.find('p').text();
    expect(textoParrafo).toBe(sub);
  })
});