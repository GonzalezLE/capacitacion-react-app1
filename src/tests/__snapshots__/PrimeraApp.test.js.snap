// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy ,luis
  </h1>
  <p>
     
  </p>
</Fragment>
`;
